sfe/sfe.go

```diff
@@ -84,23 +86,44 @@ func NewSelfServiceFrontEndImpl(
 	return sfe, nil
 }
 
+// timeoutMiddleware applies the configured per-request timeout to the request
+// context. If no timeout is set, it defaults to 5 minutes.
+func (sfe *SelfServiceFrontEndImpl) timeoutMiddleware(next http.Handler) http.Handler {
```

Comment:
[I think `http.TimeoutHandler` does this.](https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/http/server.go;l=3551)

Reply:
Ooooh, thanks!